streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -62,6 +63,18 @@
 
     private String logPrefix;
 
+    // interface to get info about the StreamThread
+    interface ThreadDataProvider {
+        InternalTopologyBuilder builder();
+        String name();
+        Set<TaskId> prevActiveTasks();
+        Set<TaskId> cachedTasks();
+        UUID processId();
+        StreamsConfig config();
+        PartitionGrouper partitionGrouper();
+        void setPartitionAssignor(final StreamPartitionAssignor assignor);
```

Comment:
Can we pass in the `TaskIdToPartitionsProvider` interface not the `StreamPartitionAssignor` here? If yes we can rename the function as well.